mesonbuild/dependencies/__init__.py

```diff
@@ -154,7 +154,9 @@ def __init__(self, name: str, environment: 'Environment', kwargs: T.Dict[str, T.
             self.is_found = False
             return
 
-        static = Mesonlib.LibType.STATIC if kwargs.get('static', False) else Mesonlib.LibType.SHARED
+        get_option = environment.coredata.get_option
+        static_opt = kwargs.get('static', get_option(Mesonlib.OptionKey('prefer_static'))
+        static = Mesonlib.LibType.STATIC if static_opt else Mesonlib.LibType.SHARED
```

Comment:
Should this change still be here? All of this is comments so I guess it doesn't matter too much, but I don't want to add wrong things in here.

Reply:
Yeah, this documentation is, in part, showing how it all fits together. So I believe it is correct to demonstrate the underlying implementation. IIRC a major motivation in adding this comment section was so that we could move from one core dev understanding how the `mesonbuild/dependencies/` directory works, to multiple core devs understanding.

After all, there is also a `self.libtype` which can technically be used too, but that doesn't obsolete the comment tutorial either.